protocols/gossipsub/src/behaviour.rs

```diff
@@ -318,6 +323,25 @@ where
         Self::new_with_subscription_filter_and_transform(
             privacy,
             config,
+            None,
+            F::default(),
+            D::default(),
+        )
+    }
+
+    /// Creates a [`Gossipsub`] struct given a set of parameters specified via a
+    /// [`GossipsubConfig`]. This has no subscription filter and uses no compression.
+    /// Metrics can be evaluated by passing a reference to a [`Registry`].
+    pub fn new_with_metrics(
+        privacy: MessageAuthenticity,
+        config: GossipsubConfig,
+        metrics_registry: &mut Registry,
+        metrics_config: MetricsConfig,
```

Comment:
Just a thought: Instead of passing both a `Registry` and ` MetricsConfig`, once could as well require the user to build `Metrics` themselves outside of `new_with_metrics` and then just passing `Metrics` to `new_with_metrics`.

Reply:
That would simplify the parameters, but would also expose a type that is otherwise useless (from an user perspective). The `Metrics` type does not expose read functions, or a fancy builder, etc. I'd prefer it this way for these reasons, what are your thoughts?

Reply:
> but would also expose a type that is otherwise useless

Good point. Might as well give users the wrong impression that they can extract the metrics from the `Metrics` struct.

Let's keep it as is.